### ISSUE TITLE
data path optimization should match exactly one schema in oneOf

### ISSUE THREAD

**Issue by timbakkum** [nrwl/nx#3240]
## Current Behavior
When running `nx build <react-app> --prod` commands with certain options through the nx console, we receive schema validation errors.

For example: 
```
> Executing task: nx build awesome-app --extractCss --extractLicenses --no-namedChunks --optimization --outputHashing=all --no-sourceMap --no-vendorChunk <


> nx run awesome-app:build --extractCss=true --extractLicenses=true --namedChunks=false --optimization=true --outputHashing=all --sourceMap=false --vendorChunk=false 
Schema validation failed with the following errors:
  Data path ".sourceMap" should be object.
  Data path ".sourceMap" should be boolean.
  Data path ".sourceMap" should match exactly one schema in oneOf.
The terminal process terminated with exit code: 1
```

Once we remove the sourceMap flag all together, the same message pops up for `optimization`.

## Expected Behavior
We expect the nx build command to run correctly when boolean flags/options are present. 

## Steps to Reproduce
Clone https://github.com/timbakkum/nx-storybook-test 

and try running the build command with one of these flags. 

### Environment
```
  @nrwl/angular : Not Found
  @nrwl/cli : 9.4.5
  @nrwl/cypress : 9.4.5
  @nrwl/eslint-plugin-nx : 9.4.5
  @nrwl/express : Not Found
  @nrwl/jest : 9.4.5
  @nrwl/linter : 9.4.5
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : Not Found
  @nrwl/react : 9.4.5
  @nrwl/schematics : Not Found
  @nrwl/tao : 9.4.5
  @nrwl/web : 9.4.5
  @nrwl/workspace : 9.4.5
  typescript : 3.8.3
```

**Comment by bekos** [comment]
hi @timbakkum :) sorry for the late reply but your issue seems to be fixed via https://github.com/nrwl/nx/commit/3cc2521 and it is available since v9.5.0. Can you confirm that is also working for you now?

**Comment by bekos** [comment]
Closing as there is no response, but feel free to open a new issue if there is still a problem
